### PR TITLE
Feature/unit tests for bed service

### PR DESCRIPTION
**Pull request by turing85**

### CODE REVIEW

File: backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,828 @@
...
+
+  @Nested
+  @DisplayName("Test calls to createNewBed")
+  class CreateNewBedTests {
+
+    @Test
+    @DisplayName("Should call dependencies and return expected result when called.")
+    void shouldCallDependenciesAndReturnExpectedResultWhenCalled() {
+      // GIVEN
+      final String state = "state";
```
Comment (jendrikjoe) [inline backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java:61]: You are reusing a vast amount of the variables in a lot of tests, namely: `state`, `type`, `ward`,` request`, `bed`, `bedType`,.... Could you move some them to fixtures, so the duplicate code is avoided? 🙂

File: backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,828 @@
...
+              null
+          ).updateBed(bedId, request)
+      );
+
+      // THEN
+      assertEquals(expectedMessage, e.getMessage());
+    }
+
+    @Test
+    @DisplayName("Should throw a InvalidBedTypeException when bed type is not found.")
```
Comment (jendrikjoe) [inline backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java]: Nitpicking here should be `an` not `a`

File: backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,828 @@
...
+              null,
+              null
+          ).updateBed(bedId, request));
+
+      // THEN
+      assertEquals(expectedMessage, e.getMessage());
+    }
+
+    @Test
+    @DisplayName("Should throw a InvalidBedStateException when bed state is not found.")
```
Comment (jendrikjoe) [inline backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java]: Nitpicking here should be an not a

File: backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java (hunk 2 above)
Comment (turing85) [reply backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java]: Done

File: backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java (hunk 1 above)
Comment (turing85) [reply backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java]: Done

File: backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java (hunk 3 above)
Comment (turing85) [reply backend/src/test/java/de/wirvsvirus/trackyourbed/BedServiceTest.java]: Done

**Pull request merged**